zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -33,3 +51,694 @@ pub(crate) fn detect_spending_accounts<'a>(
 
     Ok(acc)
 }
+
+pub(crate) fn get_transparent_receivers<P: consensus::Parameters>(
+    conn: &rusqlite::Connection,
+    params: &P,
+    account: AccountId,
+) -> Result<HashMap<TransparentAddress, Option<TransparentAddressMetadata>>, SqliteClientError> {
+    let mut ret: HashMap<TransparentAddress, Option<TransparentAddressMetadata>> = HashMap::new();
+
+    // Get all UAs derived
+    let mut ua_query = conn.prepare(
+        "SELECT address, diversifier_index_be FROM addresses WHERE account_id = :account",
+    )?;
+    let mut rows = ua_query.query(named_params![":account": account.0])?;
+
+    while let Some(row) = rows.next()? {
+        let ua_str: String = row.get(0)?;
+        let di_vec: Vec<u8> = row.get(1)?;
+        let mut di: [u8; 11] = di_vec.try_into().map_err(|_| {
+            SqliteClientError::CorruptedData("Diversifier index is not an 11-byte value".to_owned())
+        })?;
+        di.reverse(); // BE -> LE conversion
+
+        let ua = Address::decode(params, &ua_str)
+            .ok_or_else(|| {
+                SqliteClientError::CorruptedData("Not a valid Zcash recipient address".to_owned())
+            })
+            .and_then(|addr| match addr {
+                Address::Unified(ua) => Ok(ua),
+                _ => Err(SqliteClientError::CorruptedData(format!(
+                    "Addresses table contains {} which is not a unified address",
+                    ua_str,
+                ))),
+            })?;
+
+        if let Some(taddr) = ua.transparent() {
+            let index = NonHardenedChildIndex::from_index(
+                DiversifierIndex::from(di).try_into().map_err(|_| {
+                    SqliteClientError::CorruptedData(
+                        "Unable to get diversifier for transparent address.".to_string(),
+                    )
+                })?,
+            )
+            .ok_or_else(|| {
+                SqliteClientError::CorruptedData(
+                    "Unexpected hardened index for transparent address.".to_string(),
+                )
+            })?;
+
+            ret.insert(
+                *taddr,
+                Some(TransparentAddressMetadata::new(
+                    Scope::External.into(),
+                    index,
+                )),
+            );
+        }
+    }
+
+    if let Some((taddr, child_index)) = get_legacy_transparent_address(params, conn, account)? {
```

Comment:
The rebase of the first move-only commit undid the `s/child_index/address_index` rename from #1423 (41c30ef44da6e2fdbc3938148e0816a0cc704a56). Fix up the first move-only commit (and the second if necessary) to preserve the rename.